src/main/java/seedu/address/logic/commands/FindTagCommand.java

```diff
@@ -0,0 +1,35 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.TagsContainKeywordPredicate;
+
+/**
+ * Finds and lists all persons in address book whose tag contains the argument keyword.
+ * Keyword matching is case insensitive.
+ */
+public class FindTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "findtag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose tags contain "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD\n"
+            + "Example: " + COMMAND_WORD + " family ";
+
+    private final TagsContainKeywordPredicate predicate;
+
+    public FindTagCommand(TagsContainKeywordPredicate predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateFilteredPersonList(predicate);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
```

Comment:
Not too sure about this part, but it seems like the method isn't searching the list of tags for all the tags that match the predicate. Rather, it seems like the list of persons is being filtered by the predicate instead. 

Was wondering if we need to introduce an arraylist of tags associated with each user (or has it already been implemented?)

Reply:
Currently, the tags are already stored in a tag. What I did was similar to the find command which was to get the argument and compare with all the tags in the set (as shown in the ss below).
![image](https://user-images.githubusercontent.com/88677292/193464590-ddafd741-2ce1-4649-a7e7-5b4af8a5c6d7.png)